v2/yarpcclient/provider.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package yarpcclient
+
+import (
+	"go.uber.org/yarpc/v2"
+)
+
+var _ yarpc.ClientProvider = (*Provider)(nil)
+
+// Provider implements yarpc.ClientProvider.
+type Provider struct {
+	clients map[string]yarpc.Client
```

Comment:
If passing the `yarpc.Client` type by value becomes a restriction, we can update the map to be:
`map[string]*yarpc.Client`, unless it was purposefully set as `yarpc.Client` for some other reason.

Reply:
I added it as a (although minor) layer of protection so that consumers can change the map on `.Client` calls. I'd like to leave it as is (for now) if that works for you.